### PR TITLE
[Feature Store] Spark: Avoid writing empty output files

### PR DESCRIPTION
**Pull request by gtopper**
ML-1529

### CODE REVIEW

File: mlrun/feature_store/api.py [hunk 1]
```diff
@@ -816,7 +816,11 @@ def _ingest_with_spark(
             if overwrite:
                 df_to_write.write.mode("overwrite").save(**spark_options)
             else:
-                df_to_write.write.mode("append").save(**spark_options)
+                # appending an empty dataframe may cause an empty file to be created (e.g. when writing to parquet)
+                # we would like to avoid that
+                df_to_write.persist()
```
Comment (theSaarco) [inline mlrun/feature_store/api.py:821]: Why is this needed? Is this for efficiency, not having to re-calculate the `df` when saving it?

File: tests/system/feature_store/test_spark_engine.py [hunk 2]
```diff
@@ -558,3 +558,54 @@ def test_mix_of_partitioned_and_nonpartitioned_targets(self):
...
+            source,
+            run_config=fs.RunConfig(local=False),
+            targets=targets,
+            overwrite=False,
+            spark_context=self.spark_service,
+        )
+
+        # check that no files were written
+        with pytest.raises(FileNotFoundError):
+            pd.read_parquet(output_path)
```
Comment (theSaarco) [inline tests/system/feature_store/test_spark_engine.py:611]: Is this expected to ever work? Since we're now always adding `run_id` in the path, wouldn't this fail also if you do ingest real data and a file is created? I think the correct usage should be to extract the output path from the feature-set's target.

**Review by theSaarco — CHANGES_REQUESTED** [review]
Have some questions which may not be issues but just my simple misunderstanding. Let's discuss.

File: mlrun/feature_store/api.py (hunk 1 above)
Comment (gtopper) [reply mlrun/feature_store/api.py:821]: Indeed, as per
> Using cache() and persist() methods, Spark provides an optimization mechanism to store the intermediate computation of a Spark DataFrame so they can be reused in subsequent actions.

**Review by Hedingber — APPROVED** [review]
